src/Scientist/FireAndForgetResultPublisher.cs

```diff
@@ -0,0 +1,58 @@
+﻿using GitHub.Internals;
+using System;
+using System.Threading.Tasks;
+
+namespace GitHub
+{
+    public enum ResultPublisherExceptionBehavior
+    {
+        Rethrow,
+        Ignore
+    }
+
+    public class FireAndForgetResultPublisher : IResultPublisher
+    {
+        private readonly IResultPublisher _publisher;
+        private readonly Action<Exception> _onPublisherException;
+        private readonly ConcurrentSet<Task> _publishingTasks = new ConcurrentSet<Task>();
+        private readonly ResultPublisherExceptionBehavior _exceptionBehavior;
+
+        public FireAndForgetResultPublisher(IResultPublisher publisher, ResultPublisherExceptionBehavior exceptionBehavior = ResultPublisherExceptionBehavior.Ignore, Action<Exception> onPublisherException = null)
+        {
+            _publisher = publisher;
+            _onPublisherException = onPublisherException;
+            _exceptionBehavior = exceptionBehavior;
+        }
+
+        public Task Publish<T, TClean>(Result<T, TClean> result)
+        {
+
+            // Disable the warning, because the task is being tracked on the set, 
+            // and immediately removed upon completion.
+#pragma warning disable CS4014
+            var subTask = Task.Run(async () =>
+            {
+                try
+                {
+                    await _publisher.Publish(result);
+                }
+                catch(Exception ex)
+                {
+                    _onPublisherException?.Invoke(ex);
```

Comment:
We use `NullGuard` on this project so `_onPublisherException` can't be null here. If we want to give consumers the ability to pass in null for this, we need to us the `[AllowNull]` attribute on the constructor argument.